### PR TITLE
Fix bug/defects and update codes

### PR DESCRIPTION
**Pull request by 0010SS**
The following things have been updated:
1. Delete `torch.softmax()` from every MLP layer, and add one to exist only behind the final de-embed layer.
2. Clear unused libraries (left NumPy for future use) and printing statements.
3. Tie the weights of the embedding layer and the de-embedding layer.
    - This line has been commented out because for inference with a random weight setup, tying the weights will always result in the next token predicted being "fox".
4. Fix and update the readme.
5. Add mps support, which may be used for efficient training.

Nice work done today :)

### CODE REVIEW

**Review by rkdune — APPROVED** [review]
thank you!!
